### PR TITLE
`neutral_atom_compile`

### LINKED ISSUE #81: `qss` SuperstaQ provider parity
An issue to track bringing `qiskit-superstaq` up to par with our other open-source repositories. (E.g., #57, #79, #83; subsumes #55)

### PR DESCRIPTION
**Pull request by dowusu-antwi**
Adds method for `neutral_atom_compile`; closes #81.

pulser-fix-full

### CODE REVIEW

File: qiskit_superstaq/superstaq_provider.py [hunk 1]
```diff
@@ -192,3 +192,28 @@ def qscout_compile(
         from qiskit_superstaq import compiler_output
 
         return compiler_output.read_json_qscout(json_dict, circuits_list)
+
+    def neutral_atom_compile(
+        self,
+        circuits: Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]],
+        target: str = "neutral_atom_qpu",
+    ) -> Any:
```
Comment (richrines1) [inline qiskit_superstaq/superstaq_provider.py:200]: can we avoid using `Any` here? e.g.
```suggestion
    ) -> Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]]:
```
(if so it should probably be changed in cirq-superstaq as well)

**Review by vtomole — CHANGES_REQUESTED** [review]
Add a notebook similar to https://github.com/SupertechLabs/cirq-superstaq/blob/main/examples/neutral_atom_compile.ipynb

File: pulse-requirements.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+numpy==1.21.*
+pulser~=0.1.1
```
Comment (dowusu-antwi) [inline pulse-requirements.txt:2]: These requirements should be installed before running the example notebook. (If anyone has a better name for the new requirements file let me know, maybe something like `client-requirements.txt`?)

Comment (richrines1) [reply pulse-requirements.txt:2]: are these only required for `neutral_atom_compile()`? if so what about `neutral-atom-requirements.txt`?

Comment (dowusu-antwi) [reply pulse-requirements.txt]: That sounds good to me -- done.

File: qiskit_superstaq/superstaq_provider.py [hunk 3]
```diff
@@ -192,3 +192,28 @@ def qscout_compile(
...
         return compiler_output.read_json_qscout(json_dict, circuits_list)
+
+    def neutral_atom_compile(
+        self,
+        circuits: Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]],
+        target: str = "neutral_atom_qpu",
+    ) -> Union[qiskit.QuantumCircuit, List[qiskit.QuantumCircuit]]:
+        """Returns pulse schedule for the given circuit and target.
+
+        Pulse must be installed for returned object to correctly deserialize to a pulse schedule.
```
Comment (vtomole) [inline qiskit_superstaq/superstaq_provider.py:203]:
```suggestion
        Pulser must be installed for returned object to correctly deserialize to a pulse schedule.
```

**Review by vtomole — CHANGES_REQUESTED** [review]
Minor nit then LGTM

**Review by vtomole — DISMISSED** [review]
LGTM

**Review by vtomole — COMMENTED** [review]
Maybe for another PR or we can do it here: Let the user know that neutral_atom_requirements.txt` exists. Should we actually have it in the error message as well?

**Comment by vtomole** [comment]
```python
applications_superstaq.SuperstaQModuleNotFoundException(
                name=str(e.name), context="neutral_atom_compile"
            )
```
Would give
`This module wasn't found. Please pip install neutral_atom_requirements.txt` and you'll be good to go`

**Comment by vtomole** [comment]
That might be overkill. We are good with just having it in the README.

**Comment by dowusu-antwi** [comment]
Added README (ready for re-review, @vtomole @richrines1).

**Review by vtomole — APPROVED** [review]
LGTM

**Pull request merged**